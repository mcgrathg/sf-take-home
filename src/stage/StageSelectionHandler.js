import { Component } from 'react';
import PropTypes from 'prop-types';

class StageSelectionHandler extends Component {
  static propTypes = {
    children: PropTypes.func.isRequired,
  };

  state = {
    selected: {},
  };

  onOptionToggled = (option = '', isToggled = true) => {
    this.setState(({ selected }) => ({
      selected: { ...selected, [option]: isToggled },
    }));
  };

  resetSelections = () => {
    this.setState({ selected: {} });
  };

  render() {
    const { children } = this.props;
    const { selected } = this.state;

    return children({
      onOptionToggled: this.onOptionToggled,
      resetSelections: this.resetSelections,
      selected,
    });
  }
}

export default StageSelectionHandler;
