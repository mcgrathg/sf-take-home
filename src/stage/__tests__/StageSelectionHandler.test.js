import React from 'react';
import { render } from 'react-testing-library';
import { childrenFunc } from 'test-utils';
import StageSelectionHandler from '../StageSelectionHandler';

describe('StageSelectionHandler', () => {
  it('renders', () => {
    const { container } = render(
      <StageSelectionHandler>{childrenFunc}</StageSelectionHandler>,
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
