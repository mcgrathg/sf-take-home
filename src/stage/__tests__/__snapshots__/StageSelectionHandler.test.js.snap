// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StageSelectionHandler renders 1`] = `
<div>
  something
</div>
`;
